From 8d5889209bee330facbb66801f7f5056e3910af8 Mon Sep 17 00:00:00 2001
From: Antonios Vamporakis <ant@area128.com>
Date: Mon, 6 Jan 2014 19:18:09 +0100
Subject: tools: ltq-boot-image: allow numerical input in any base

strtoul is smart enough to read base 8, 10 or 16 numbers. This update allows
it to be smart and adds certain freedom in the formating of config files.

Signed-off-by: Antonios Vamporakis <ant@area128.com>

diff --git a/tools/ltq-boot-image.c b/tools/ltq-boot-image.c
index 743fc6f..1f1e3a9 100644
--- a/tools/ltq-boot-image.c
+++ b/tools/ltq-boot-image.c
@@ -89,13 +89,13 @@ static int parse_args(int argc, char *argv[], struct args *arg)
 			arg->type = parse_image_type(optarg);
 			break;
 		case 'e':
-			arg->entry_addr = strtoul(optarg, NULL, 16);
+			arg->entry_addr = strtoul(optarg, NULL, 0);
 			break;
 		case 'x':
-			arg->uboot_offset = strtoul(optarg, NULL, 16);
+			arg->uboot_offset = strtoul(optarg, NULL, 0);
 			break;
 		case 'p':
-			arg->page_size = strtoul(optarg, NULL, 10);
+			arg->page_size = strtoul(optarg, NULL, 0);
 			break;
 		case 'u':
 			arg->uboot_bin = optarg;
-- 
2.7.4

